experimental/relativetime/src/format.rs

```diff
@@ -0,0 +1,85 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::fmt::Write;
+
+use fixed_decimal::{FixedDecimal, Sign};
+use writeable::Writeable;
+
+use crate::{
+    options::{Numeric, RelativeTimeFormatterOptions},
+    relativetime::RelativeTimeFormatter,
+};
+
+pub mod parts {
+    use writeable::Part;
+
+    /// The [`Part`] used by [`FormattedRelativeTime`](crate::FormattedRelativeTime) to mark the
+    /// part of the string that is without a placeholder.
+    pub const LITERAL: Part = Part {
+        category: "relativetime",
+        value: "literal",
+    };
+}
+
+/// An intermediate structure returned by [`RelativeTimeFormatter`](crate::RelativeTimeFormatter).
+/// This structure can be consumed via [`Writeable`](Writeable) trait to a string or buffer.
+pub struct FormattedRelativeTime<'a> {
+    pub(crate) formatter: &'a RelativeTimeFormatter,
+    pub(crate) options: &'a RelativeTimeFormatterOptions,
+    pub(crate) value: FixedDecimal,
+}
+
+impl<'a> Writeable for FormattedRelativeTime<'a> {
+    fn write_to_parts<S: writeable::PartsWrite + ?Sized>(&self, sink: &mut S) -> core::fmt::Result {
+        if self.options.numeric == Numeric::Auto {
+            let _ = &self.formatter.rt.get().relatives;
+        }
+
+        let plural_rules_mapping = if self.value.sign() == Sign::Negative
```

Comment:
This is `A || (A && B)`, so `B` doesn't matter. Typo?

Reply:
Ah yeah fixed this.